arthur/tasks.py

```diff
@@ -228,26 +235,67 @@ def get(self, task_id):
         :raises NotFoundError: raised when the requested task is not
             found on the registry
         """
-        try:
-            self._rwlock.reader_acquire()
-            task = self._tasks[task_id]
-        except KeyError:
+        self._rwlock.reader_acquire()
+        found = self.conn.exists(task_id)
+        self._rwlock.reader_release()
+
+        if not found:
             raise NotFoundError(element=str(task_id))
-        finally:
-            self._rwlock.reader_release()
+
+        self._rwlock.reader_acquire()
+        task_dump = self.conn.get(task_id)
+        task = pickle.loads(task_dump)
+        self._rwlock.reader_release()
```

Comment:
Here there's a race condition. A writer can remove the data between the two blocks which will make you to have an empty task object.